recipes/rust/build.sh

```diff
@@ -4,7 +4,19 @@
 make
 make install
 
+
+# Download a precompiled version of cargo.
+# This is needed to build cargo itself.
+if [[ $OSTYPE == darwin* ]]
+then
+    CARGO_STAGE0=https://static.rust-lang.org/dist/cargo-$PKG_VERSION-x86_64-apple-darwin.tar.gz
+else
+    CARGO_STAGE0=https://static.rust-lang.org/dist/cargo-$PKG_VERSION-x86_64-unknown-linux-gnu.tar.gz
+fi
+
+curl $CARGO_STAGE0 | tar xf cargo-0.16.0-x86_64-unknown-linux-gnu.tar.gz */cargo/bin/cargo --strip=2
+
```

Comment:
The thing is that rustc and cargo building need a rust and cargo compiler (the so-called stage0) themselves. Here, I tried to obtain that from a separate binary, but it turns out that this is already baked into the rustc makefile. Seems like I just have to point the cargo build to stage0. Let's see if that works.